common/src/main/java/net/opentsdb/query/filter/NotFilter.java

```diff
@@ -57,6 +59,28 @@ public String getType() {
   public static Builder newBuilder() {
     return new Builder();
   }
+
+  @Override
+  public int hashCode() {
+    return buildHashCode().asInt();
+  }
+
+  /** @return A HashCode object for deterministic, non-secure hashing */
+  public HashCode buildHashCode() {
+    return filter.buildHashCode();
```

Comment:
We need to add something here to differentiate the Not from the underlying filter. E.g. hashcode list where you just add the class name to the hash or something.